library/Faktory/Worker.hs

```diff
@@ -90,8 +91,12 @@ processorLoop client settings workerSettings f = do
   let
     namespace = connectionInfoNamespace $ settingsConnection settings
     processAndAck job = do
-      f job
-      ackJob client job
+      mResult <- timeout (getJobTTL job) $ do
+        f job
+        ackJob client job
+      case mResult of
+        Nothing -> settingsLogError settings "Job reservation period expired."
+        Just{} -> pure ()
```

Comment:
Completely minor, but I might do it this way, just to ensure if we ever changed the type of `f` / `ackJob`, we'd be alerted to this point in the code about it.

```suggestion
        Just () -> pure ()
```